var/spack/repos/builtin/packages/quantum-espresso/package.py

```diff
@@ -280,6 +329,16 @@ def install(self, spec, prefix):
 
         configure(*options)
 
+        # Filter file must be applied after configure executes
+        # QE 6.4.0 to QE 6.4 have `-L` missing in front of zlib library
+        if spec.variants['hdf5'].value != 'none':
+            if spec.satisfies('@6.1.0:6.4.0'):
+                make_inc = join_path(self.stage.source_path, 'make.inc')
+                zlib_libs = spec['zlib'].prefix.lib + ' -lz'
```

Comment:
I would use `spec['zlib'].libs.ld_flags` since the library may be in `lib` or `lib64`

Reply:
Code is a bit confusing here. Line 337 contains what comes out of QE's configure, which is wrong. QE's configure yields the following in `make.inc` :
`<spack prefix>/zlib-<stuff>/lib -lz` (it is missing the `-L` in front)

On Lines 338-340, I apply a filter_file to `zlib_libs` variable and replace it with exactly what you are suggesting.